website/docs/cloud-docs/integrations/kubernetes/ops-v2-migration.mdx

```diff
@@ -43,33 +43,220 @@ Next, backup the workspace resources.
+| `spec.module.version` | `spec.module.version` | Refer to  [module sources](https://developer.hashicorp.com/terraform/language/modules/sources) for versioning information for each module source. |
+| `spec.variables.[*]` | `spec.variables.[*].name` | Variable names should be included in the module. This is a reference to variables in the Workspace where the module will be executed. |
+| `spec.outputs.[*].key` | `spec.outputs.[*].name` | Output names should be included in the module. This is a reference to the output variables produced by the module. |
+| `status.workspaceID` OR `metadata.namespace-metadata.name` | `spec.workspace.id` OR `spec.workspace.name` | The workspace where the module will be executed. The workspace must be within the same organization. |
+
+Below is an example migration of a `Module` between v1 and v2 of the operator:
+
+<CodeBlockConfig filename="v1.yaml">
+
+```yaml
+apiVersion: app.terraform.io/v1alpha1
+kind: Workspace
+metadata:
+  name: migration
+spec:
+  module:
+    source: app.terraform.io/org-name/module-name/provider
+    version: 0.0.42
+  variables:
+    - key: username
+      value: "user"
+      hcl: true
+      sensitive: false
+      environmentVariable: false
+    - key: SECRET_KEY
+      value: "s3cr3t"
+      hcl: false
+      sensitive: false
+      environmentVariable: true
+```
+
+</CodeBlockConfig>
+
+In v2 of the operator, the workspace and module are manage by separate controllers.
+
+<CodeBlockConfig filename="workspace-v2.yaml">
+
+  ```yaml
+  apiVersion: app.terraform.io/v1alpha2
+  kind: Workspace
+  metadata:
+    name: migration
+    spec:
+      terraformVariables:
+        - name: username
+          value: "user"
+          hcl: true
+          sensitive: false
+      environmentVariables:
+        - name: SECRET_KEY
+          value: "s3cr3t"
+          hcl: false
+          sensitive: false
+  ```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig filename="module-v2.yaml">
+
+```yaml
+apiVersion: app.terraform.io/v1alpha2
+kind: Module
+metadata:
+  name: migration
+spec:
+  name: operator
+  module:
+    source: app.terraform.io/org-name/module-name/provider
+    version: 0.0.42
+  workspace:
+    name: migration
+```
+
+</CodeBlockConfig>
+
+## Upgrade the operator
+
+Download Workspace CRD patch A:
 
 ```shell-session
-$ cd terraform-cloud-operator/docs/migration/crds
+$ curl -sO https://raw.githubusercontent.com/hashicorp/terraform-cloud-operator/main/docs/migration/crds/workspaces_patch_a.yaml
 ```
 
 View the changes that **patch A** will apply to the workspace CRD.
 
 ```shell-session
-$ kubectl diff -f workspaces_patch_a.yaml
+$ kubectl diff --filename workspaces_patch_a.yaml
 ```
 
 Patch the workspace CRD with **patch A**. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON unmarshalling issues.
```

Comment:
```suggestion
Patch the workspace CRD with patch A. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON un-marshalling issues.
```